sentry_sdk/integrations/ariadne.py

```diff
@@ -0,0 +1,175 @@
+    from graphql.language.ast import DocumentNode  # type: ignore
+    from sentry_sdk._types import EventProcessor
+
+
+class AriadneIntegration(Integration):
+    identifier = "ariadne"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        installed_packages = _get_installed_modules()
+        version = parse_version(installed_packages["ariadne"])
+
+        if version is None:
+            raise DidNotEnable("Unparsable ariadne version: {}".format(version))
+
+        if version < (0, 20):
+            raise DidNotEnable("ariadne 0.20 or newer required.")
+
+        _patch_graphql()
+
+
+def _patch_graphql():
+    # type: () -> None
+    old_parse_query = ariadne_graphql.parse_query
+    old_handle_errors = ariadne_graphql.handle_graphql_errors
+    old_handle_query_result = ariadne_graphql.handle_query_result
+
+    def _sentry_patched_parse_query(context_value, query_parser, data):
+        # type: (Optional[Any], Optional[QueryParser], Any) -> DocumentNode
+        hub = Hub.current
+        integration = hub.get_integration(AriadneIntegration)
+        if integration is None:
+            return old_parse_query(context_value, query_parser, data)
+
+        with hub.configure_scope() as scope:
+            event_processor = _make_request_event_processor(data)
+            scope.add_event_processor(event_processor)
+
+        result = old_parse_query(context_value, query_parser, data)
+        return result
+
+    def _sentry_patched_handle_graphql_errors(errors, *args, **kwargs):
+        # type: (List[GraphQLError], Any, Any) -> GraphQLResult
+        hub = Hub.current
+        integration = hub.get_integration(AriadneIntegration)
+        if integration is None:
+            return old_handle_errors(errors, *args, **kwargs)
+
+        result = old_handle_errors(errors, *args, **kwargs)
+
+        with hub.configure_scope() as scope:
+            event_processor = _make_response_event_processor(result[1])
+            scope.add_event_processor(event_processor)
+
+        if hub.client:
+            with capture_internal_exceptions():
+                for error in errors:
+                    event, hint = event_from_exception(
+                        error,
+                        client_options=hub.client.options,
+                        mechanism={
+                            "type": hub.get_integration(AriadneIntegration).identifier,
+                            "handled": False,
+                        },
+                    )
+                    hub.capture_event(event, hint=hint)
+
+        return result
+
+    def _sentry_patched_handle_query_result(result, *args, **kwargs):
+        # type: (Any, Any, Any) -> GraphQLResult
+        hub = Hub.current
+        integration = hub.get_integration(AriadneIntegration)
+        if integration is None:
+            return old_handle_query_result(result, *args, **kwargs)
+
+        response = old_handle_query_result(result, *args, **kwargs)
```

Comment:
Nit: in the other functions it is called `result` and here `response`. if on purpose ok.

Reply:
The first arg of the function is called `result`, so can't use that here, changed `response` to `query_result` at least.